tools/sdk/ld/eagle.app.v6.common.ld.h

```diff
@@ -127,8 +127,8 @@ SECTIONS
     *(.init.literal)
     *(.init)
 
-    /* Special functions/templates that need to be in IRAM (SPI/IRQ callbacks/etc. here */
-    *(.text._ZNKSt8functionIFvvEEclEv)  /* std::function<void ()>::operator()() const */
+    /* all functional callers are placed in IRAM (including SPI/IRQ callbacks/etc) here */
+    *(.text._ZNKSt8functionIF*EE*)  /* std::function<any(...)>::operator()() const */
```

Comment:
I think this stops evaluation at the first asterisk.  The regular expression support is very basic in the linker.

Reply:
I quickly got lost in the big yacc parser.
So to verify I tweaked a test file object,
Before:
```
$ nm /tmp/arduino_build_491121/function.ino.elf|grep _ZNKSt8functio
401000b0 W _ZNKSt8functionIFviEEclEi
40100094 W _ZNKSt8functionIFvvEEclEv
```
Then I edited the object file and replaced `_ZNKSt8functionIFvvEEclEv` by `_ZNKSt8functionIFvvEXclEv` (EE->EX after the first asterisk), then relinked:
```
$ nm /tmp/arduino_build_491121/function.ino.elf|grep _ZNKSt8functio
40100094 W _ZNKSt8functionIFviEEclEi
402010c4 W _ZNKSt8functionIFvvEXclEv
```

Is this a sufficient validation ?

Reply:
Yes, that looks fine.